src/api/viewRows.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { defaultFetch } from 'fetching';
+import handleResponseData from './utils/handleResponseData';
+import { makeUseMutationOptions } from './utils/resourceHookFactories';
+import { useMutation, useQueryClient } from 'react-query';
+import { ZetkinPerson, ZetkinViewRow } from 'types/zetkin';
+
+
+export const viewRowsResource = (orgId: number, viewId: string) => {
```

Comment:
@djbusstop  So this is the way of doing this recommended by https://react-query.tanstack.com/guides/updates-from-mutation-responses
So that's good - the question is, can this be abstracted? So that we can end up with something closer to:

`useDelete: createUseMutationDelete(key, url),`